lib/spack/spack/repo.py

```diff
@@ -69,14 +69,14 @@ def get_full_namespace(namespace):
 #: core the top-level namespace polluted by package symbols this way.  To
 #: solve this, the top-level ``spack`` package contains very few symbols
 #: of its own, and importing ``*`` is essentially a no-op.  The common
-#: routines and directives that packages need are now in ``spack.pkgkit``,
+#: routines and directives that packages need are now in ``spack.package``,
 #: and the import system forces packages to automatically include
 #: this. This way, old packages that call ``from spack import *`` will
 #: continue to work without modification, but it's no longer required.
 #:
 #: TODO: At some point in the future, consider removing ``from spack import *``
-#: TODO: from packages and shifting to from ``spack.pkgkit import *``
-_package_prepend = 'from spack.pkgkit import *'
+#: TODO: from packages and shifting to from ``spack.package import *``
+_package_prepend = 'from spack.package import *'
```

Comment:
I would be in favor of removing this logic entirely and forcing packages to include `from spack.package import *`. At the very least, we could deprecate it if we're worried about non-builtin packages. How much of a performance improvement will this give us?

Reply:
Me too, but the idea was to leave it deprecated for a couple of versions for packages outside of the builtin repo.

I'm not sure on performance.  It depends on how much other stuff we can refactor out with the prepend removed I think, since the prepend is not super slow in and of itself, but having to load the whole thing and modify it rather than direct load it probably has a non-trivial cost, especially in that it can't be cached.